File: backend/platform-data/src/main/java/com/platform/common/entity/Comment.java
/**
 * @author HXN
 * @date 2026-08-30
 * @description 评论实体类
 */
package com.platform.common.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 评论实体
 *
 * <p>对应数据库 comment 表。支持按业务类型（biz_type）+ 业务对象 ID（biz_id）
 * 进行通用关联，可用于需求条目、手动化用例等业务对象的评论功能。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("comment")
public class Comment extends BaseEntity {

    private static final long serialVersionUID = 1L;

    /**
     * 业务类型：REQUIREMENT_ITEM-需求条目，MANUAL_CASE-手动化用例
     */
    private String bizType;

    /**
     * 业务对象 ID
     */
    private Long bizId;

    /**
     * 评论内容
     */
    private String content;

    /**
     * 父评论 ID（null 表示一级评论）
     */
    private Long parentId;

    /**
     * 评论人 ID
     */
    private Long createdBy;
}

File: frontend/src/api/comment.ts
/**
 * @author HXN
 * @date 2026-08-30
 * @description 评论模块 API
 */
import request from './request'

export interface CommentItem {
  id: number
  bizType: string
  bizId: number
  content: string
  parentId: number | null
  createdBy: number | null
  createdByName: string | null
  createdAt: string
  children?: CommentItem[]
}

export interface CommentCreateRequest {
  bizType: string
  bizId: number
  content: string
  parentId?: number | null
}

/** 查询业务对象下的评论列表 */
export function getComments(bizType: string, bizId: number) {
  return request.get('/v1/comments', { params: { bizType, bizId } })
}

/** 发表评论/回复 */
export function createComment(data: CommentCreateRequest) {
  return request.post('/v1/comments', data)
}

/** 删除评论 */
export function deleteComment(commentId: number) {
  return request.post(`/v1/comments/${commentId}/delete`)
}
